### PR TITLE
Vk protected content final

### PR DESCRIPTION
**Pull request by phoenixxxx**
This is the last part of the protected contend feature:

- Cache the render buffer in the render pass strucutre
- Enable the proteded path in the Render Target
- Make sure we build the protected version of the depth/color in the swap chaing
- Minor cleanup/restrucuring of the code

### CODE REVIEW

File: filament/backend/src/vulkan/VulkanBlitter.cpp [hunk 1]
```diff
@@ -151,7 +151,8 @@ void VulkanBlitter::resolve(VulkanAttachment dst, VulkanAttachment src) {
     }
 #endif
 
-    VulkanCommandBuffer& commands = mCommands->get();
+    VulkanCommandBuffer& commands = dst.texture->getIsProtected() ?
+    mCommands->getProtected() : mCommands->get();
```
Comment (poweifeng) [inline filament/backend/src/vulkan/VulkanBlitter.cpp:155]: 8 spaces here

File: filament/backend/src/vulkan/platform/VulkanPlatformSwapChainImpl.cpp [hunk 2]
```diff
@@ -110,8 +115,8 @@ void VulkanPlatformSwapChainImpl::destroy() {
     mSwapChainBundle.colors.clear();
 }
 
-VkImage VulkanPlatformSwapChainImpl::createImage(VkExtent2D extent, VkFormat format) {
-    auto [image, memory] = createImageAndMemory(mContext, mDevice, extent, format);
+VkImage VulkanPlatformSwapChainImpl::createImage(VkExtent2D extent, VkFormat format, bool isProtected) {
```
Comment (poweifeng) [inline filament/backend/src/vulkan/platform/VulkanPlatformSwapChainImpl.cpp:118]: line too long

File: filament/backend/src/vulkan/platform/VulkanPlatformSwapChainImpl.cpp [hunk 3]
```diff
@@ -248,7 +253,7 @@ VkResult VulkanPlatformSurfaceSwapChain::create() {
     mSwapChainBundle.colorFormat = surfaceFormat.format;
     mSwapChainBundle.depthFormat =
             selectDepthFormat(mContext.getAttachmentDepthStencilFormats(), mHasStencil);
-    mSwapChainBundle.depth = createImage(mSwapChainBundle.extent, mSwapChainBundle.depthFormat);
+    mSwapChainBundle.depth = createImage(mSwapChainBundle.extent, mSwapChainBundle.depthFormat, mIsProtected);
```
Comment (poweifeng) [inline filament/backend/src/vulkan/platform/VulkanPlatformSwapChainImpl.cpp:256]: line too long

**Comment by phoenixxxx** [comment]
Feedback addressed.

**Review by poweifeng — APPROVED** [review]

**Pull request merged**